### ISSUE TITLE
Upgrading from 1.2.4 to 1.2.5 now requires me to exclude FreeMarkerConfig

### ISSUE THREAD

**Issue by pgcarter** [spring-projects/spring-boot#3418]
When I upgraded to 1.2.5 freemarker dependency is being dragged into my application project.

I now need to explicitly exclude the configuration:

@EnableAutoConfiguration(exclude = FreeMarkerAutoConfiguration.class)

**Comment by snicoll** [comment]
Please provide more details. Do you see that the `freemarker` **dependency** is being dragged or do you see any kind of error?

**Comment by pgcarter** [comment]
The dependency is being dragged in. I started to get failures on startup because it could not find the template directory. Before the upgrade the FreeMarkerAutoConfiguration was not called because freemarker was not a dependency.

**Comment by snicoll** [comment]
That still doesn't tell me who's dragging it. Can you run `mvn dendency:tree` or `gradlew dependencies` to find out?

**Comment by pgcarter** [comment]
Sorry did a Hazelcast upgrade at same time that's what was bringing in Freemarker
+- com.hazelcast:hazelcast-all:jar:3.5:compile
[INFO] |  +- net.sourceforge.findbugs:annotations:jar:1.3.2:compile
[INFO] |  +- com.eclipsesource.minimal-json:minimal-json:jar:0.9.2:compile
[INFO] |  \- org.freemarker:freemarker:jar:2.3.22:compile

**Comment by GreyTeardrop** [comment]
I'm having the same issue with Spring Boot 1.3.0.M1 -> 1.3.0.M2 upgrade.

My application had transitive dependency on com.hazelcast:hazelcast-client:jar:3.3.2. Upgrading to spring-boot-starter-parent 1.3.0.M2 (with no other changes) enforces newer version of Hazelcast (com.hazelcast:hazelcast:jar:3.5) which in turn now has transitive dependency on org.freemarker:freemarker:jar:2.3.22. That causes application startup to fail with 

```
Caused by: java.lang.IllegalArgumentException: Cannot find template location(s): [classpath:/templates/] (please add some templates, check your FreeMarker configuration, or set spring.freemarker.checkTemplateLocation=false)
	at org.springframework.util.Assert.notNull(Assert.java:115)
	at org.springframework.boot.autoconfigure.freemarker.FreeMarkerAutoConfiguration.checkTemplateLocationExists(FreeMarkerAutoConfiguration.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:354)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:305)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:133)
```

The workaround is provided in exception message and does work (by the way, that's great exception message!), but it's somewhat confusing nevertheless, as none of my "own" dependencies bring Freemarker to classpath to begin with.

**Comment by snicoll** [comment]
We will provide an explicit exclude for that in RC1, thanks for bringing this to our attention.

**Comment by snicoll** [comment]
I've added an exclusion that will be available in `1.3.0.RC1` in case Hazelcast 3.5.1 is not released by then. @pgcarter  you are affected by the issue but in a different way since the hazelcast support was only introduced in the 1.3 line. I'd advise you to add an exclusion on the freemarker dependency as I did in https://github.com/spring-projects/spring-boot/commit/1348acf3e3faa6fe325de9abcad5b1c46e15e3d7

**Comment by GreyTeardrop** [comment]
Thank you!
